include/votca/csg/beadstructure.h

```diff
@@ -76,27 +75,28 @@ class BeadStructure {
    *
    * The same bead cannot be added twice.
    **/
-  void AddBead(T *bead);
+  void AddBead(T &bead);
```

Comment:
I am not truly happy with that interface, because it does not make the ownership clear. Who owns bead?

Reply:
What would you do instead? Shared pointer?

Reply:
Use a copy?

Reply:
Does bead structure own beads or is it more a view on a collection of beads?

Reply:
I do not know

Reply:
It was meant to act as a window, but I have changed it now so that it does not store pointers internally. This means one has to now be careful that if the beads change that they use a new beadstructure. Though only the masses and names of the beads are stored internally, and these are very unlikely to change.

Reply:
window=view? Can you explain a bit more?